test/byContract/registries/registries.test.ts

```diff
@@ -0,0 +1,452 @@
+        });
+    });
+
+    describe('setBaseURI', () => {
+        it('Should retrieve base URI', async () => {
+            const chainId = hre.network.config.chainId;
+            expect(await appRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/app/${chainId}/`,
+            );
+            expect(await datasetRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/dataset/${chainId}/`,
+            );
+            expect(await workerpoolRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/workerpool/${chainId}/`,
+            );
+        });
+
+        it('Should not set base URI when user is not the owner', async () => {
+            await expect(appRegistry.setBaseURI(`https://new.url.iex.ec/app/`)).to.be.revertedWith(
+                'Ownable: caller is not the owner',
+            );
+            await expect(
+                datasetRegistry.setBaseURI(`https://new.url.iex.ec/dataset/`),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+            await expect(
+                workerpoolRegistry.setBaseURI(`https://new.url.iex.ec/workerpool/`),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+        });
+    });
+
+    describe('App Registry', () => {
+        const createAppArgs = [
+            `App`,
+            'DOCKER',
+            constants.MULTIADDR_BYTES,
+            ethers.utils.id(`Content of my app`),
+            '0x1234',
+        ] as [string, string, BytesLike, BytesLike, BytesLike];
+        it('Should predict the correct address for future app creation', async () => {
+            const code = await appRegistry.proxyCode();
+            const proxyCodeHash = ethers.utils.keccak256(code);
+            const encodedInitializer = App__factory.createInterface().encodeFunctionData(
+                'initialize',
+                createAppArgs,
+            );
+            const salt = ethers.utils.solidityKeccak256(
+                ['bytes', 'address'],
+                [encodedInitializer, appProvider.address],
+            );
+
+            const predictedAddress = ethers.utils.getCreate2Address(
+                appRegistry.address,
+                salt,
+                proxyCodeHash,
+            );
+            expect(await appRegistry.predictApp(appProvider.address, ...createAppArgs)).to.equal(
+                predictedAddress,
+            );
+        });
+
+        it('Should create the app', async () => {
+            const predictedAddress = await appRegistry.predictApp(
+                appProvider.address,
+                ...createAppArgs,
+            );
+            await expect(await appRegistry.createApp(appProvider.address, ...createAppArgs))
+                .to.emit(appRegistry, 'Transfer')
+                .withArgs(
+                    AddressZero,
+                    appProvider.address,
+                    ethers.BigNumber.from(predictedAddress).toString(),
```

Comment:
I think here is a good place to check, don't you?
```
        _holderTokens[to].add(tokenId);
        _tokenOwners.set(tokenId, to);
```

Reply:
is this not testing the ERC721 standard ? I can add it yes

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/122/commits/340735c645cc4a8c9f311561a163d068810b9174)

Reply:
IMO we should test when possible all behaviors of a contract, for instance we need to check that _mintCreate/_mint are called on createApp. I hope you & @zguesmi agree (?)
What you added is good to me :+1:
Though if I'm not wrong checks are missing for dataset & workerpool (?)

Reply:
On my way!